testbed/tests/test_font.py

```diff
@@ -0,0 +1,131 @@
+import sys
+from pathlib import Path
+from typing import Any
+
+import pytest
+
+import toga
+from toga.fonts import BOLD, ITALIC, SYSTEM, Font
+
+from .conftest import skip_on_platforms
+
+
+@pytest.fixture
+async def widget() -> toga.Label:
+    skip_on_platforms("android", "iOS", "macOS")
+    return toga.Label("hello, this is a label")
+
+
+@pytest.fixture
+async def app_path(app):
+    # This *should* be app.paths.app, but under test conditions,
+    # the mainline is the test module, which confuses the app path.
+    return Path(sys.modules[app.__module__].__file__).parent
+
+
+async def test_use_system_font_fallback(
+    widget: toga.Label,
+    widget_probe: Any,
+    capsys: pytest.CaptureFixture[str],
+):
+    """If an unknown font is requested, the system font is used as a fallback."""
+    widget_probe.assert_font_family(SYSTEM)
+    widget.style.font_family = "unknown"
+    await widget_probe.redraw("Falling back to system font")
+
+    assert "using system font as a fallback" in capsys.readouterr().out
+
+
+async def test_font_options(widget: toga.Label, widget_probe: Any):
+    """Every combination of weight, style and variant can be used on a font."""
+    for font_weight in widget_probe.FONT_WEIGHTS:
+        for font_style in widget_probe.FONT_STYLES:
+            for font_variant in widget_probe.FONT_VARIANTS:
```

Comment:
This approach doesn't test that a backend correctly ignores the values that it doesn't support. 

Instead, how about using the same approach as widgets/test_label.py::test_alignment – always testing every combination of values, but giving the probe an assertion method which can verify how unsupported values are handled.

Reply:
Good point - I hadn't considered taking it from that angle. I'll update the test + probes.